pandas/core/sorting.py

```diff
@@ -255,15 +254,14 @@ def nargsort(items, kind='quicksort', ascending=True, na_position='last'):
             sorted_idx = np.roll(sorted_idx, cnt_null)
         return sorted_idx
 
-    with warnings.catch_warnings():
-        # https://github.com/pandas-dev/pandas/issues/25439
-        # can be removed once ExtensionArrays are properly handled by nargsort
-        warnings.filterwarnings(
-            "ignore", category=FutureWarning,
-            message="Converting timezone-aware DatetimeArray to")
+    if (not isinstance(items, ABCIndexClass)
```

Comment:
One more question: why do we have this check for ABCIndexClass here?  Shouldn't they be treated the same?

I think it'd be more correct to do

```
from pandas.core.internals.construction import extract_array

items = extract_array(items)
```

on all inputs. This will get an ExtensionArray from an Index or Series. Then if the new `items` is_extension_array_dtype, we do the `_values_for_argsort`.

Reply:
`extract_array` is imported inside `nargsort` to avoid circular import.